### PR TITLE
Shorten tests

### LINKED ISSUE #751: Optimize CI tests
Some of `freud`'s CI tests take are expensive and slow (~5.5 min). We should review and edit tests for performance improvement.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This PR reduces the runtime of most of the most expensive tests in freud. On my Mac, this reduces the total time for tests from ~5:15 to ~1:30. The remaining expensive tests are I think justified in their length: environment matching with global search, the RDF ideal gas test, and Minkowski structure metrics account for roughly half the time, but are important and not easy to simplify further.

## Motivation and Context

Resolves: #751 

## How Has This Been Tested?

Existing tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

I don't think this needs a changelog entry or an update to the credits.

### CODE REVIEW

File: tests/test_density_LocalDensity.py [hunk 1]
```diff
@@ -42,22 +41,23 @@ def test_density(self):
...
+        self.ld.box
 
-            assert self.ld.box == freud.box.Box.cube(10)
+        assert self.ld.box == freud.box.Box.cube(10)
 
-            npt.assert_array_less(np.fabs(self.ld.density - 10.0), 1.5)
+        npt.assert_array_less(np.fabs(self.ld.density - 10.0), 1.5)
 
-            npt.assert_array_less(np.fabs(self.ld.num_neighbors - 1130.973355292), 200)
+        npt.assert_array_less(np.fabs(self.ld.num_neighbors - 1130.973355292), 200)
```
Comment (bdice) [inline tests/test_density_LocalDensity.py:60]: Can we add a comment for the magic numbers in this test? I have no idea where they come from.

Comment (vyasr) [reply tests/test_density_LocalDensity.py:60]: I would love to, but I don't know either. I'd assume from the use of `np.fabs` that this is a regression test Harper wrote way back in the day.

File: tests/test_density_RDF.py [hunk 2]
```diff
@@ -79,34 +79,36 @@ def test_random_point(self):
...
+            nq = freud.locality.AABBQuery(box, points)
+            neighbors = {"mode": "ball", "r_max": r_max, "exclude_ii": True}
+
+            rdf = freud.density.RDF(bins, r_max, r_min)
+
+            if r_min != 3.0:
+                rdf.compute(nq, neighbors=neighbors, reset=False)
+            else:
+                rdf.compute(nq, neighbors=neighbors)
+
```
Comment (SchoeniPhlippsn) [inline tests/test_density_RDF.py:96]: You removed the assertion which compares `rdf.box`. Is there a specific reason behind that?

**Review by bdice — APPROVED** [review]
I approve once @SchoeniPhlippsn's eagle-eyed comment about `rdf.box` is resolved. This can be merged after that is handled.

**Comment by bdice** [comment]
I need this PR to be merged to speed up my work on #749. I'm going to revert @vyasr's change that removed the box assertion. If tests pass, I will merge this PR.

File: tests/test_density_RDF.py [hunk 3]
```diff
@@ -79,34 +79,36 @@ def test_random_point(self):
...
+            # inputs, so we only test the fastest one (AABBQuery).
+            nq = freud.locality.AABBQuery(box, points)
+            neighbors = {"mode": "ball", "r_max": r_max, "exclude_ii": True}
+
+            rdf = freud.density.RDF(bins, r_max, r_min)
+
+            if r_min != 3.0:
+                rdf.compute(nq, neighbors=neighbors, reset=False)
+            else:
+                rdf.compute(nq, neighbors=neighbors)
```
Comment (bdice) [inline tests/test_density_RDF.py:95]:
```suggestion
                rdf.compute(nq, neighbors=neighbors)
            assert rdf.box == box
```

File: tests/test_density_RDF.py (hunk 2 above)
Comment (bdice) [reply tests/test_density_RDF.py:96]:
```suggestion

            assert rdf.box == box
```

**Comment by bdice** [comment]
Ouch, that's not good -- the CircleCI tests passed but the cibuildwheel tests failed. Since cibuildwheel isn't a required check (yet?), the PR was auto-merged. We'll need to keep an eye on this in case more failures occur. This could have been a one-off error, but it would be a big problem for our release deployments if tests fail nondeterministically.

```
  =================================== FAILURES ===================================
  _____________________________ TestPMFTXYZ.test_pcf _____________________________
  [gw0] linux -- Python 3.7.9 /tmp/tmp.bbbU5gf5RF/venv/bin/python
  
  self = <test_pmft.TestPMFTXYZ object at 0x7f9da4592290>
  
      def test_pcf(self):
          """Verify that integrating a PMFT to generate an RDF for an ideal gas
          produces approximately unity everywhere."""
      
          def get_pmft_bydist(r_max, nbins):
              """Get a PMFT with a specified radial cutoff."""
              limit = np.sqrt(r_max ** 2 / self.ndim)
              return self.pmft_cls(*(limit,) * len(self.limits), bins=nbins)
      
          L = 10
          N = 2500
          nbins = 100
          r_max = 3
      
          system = freud.data.make_random_system(L, N, self.ndim == 2, seed=3)
          orientations = (
              rowan.random.rand(N) if self.ndim == 3 else np.random.rand(N) * 2 * np.pi
          )
      
          pmft = get_pmft_bydist(r_max, nbins)
          pmft.compute(system, orientations)
      
          def get_bin_centers(pmft):
              if len(self.limits) > 1:
                  return pmft.bin_centers[: len(self.limits)]
              else:
                  return [pmft.bin_centers[0]]
      
          radii = build_radii(get_bin_centers(pmft))
          rdf = pmft_to_rdf(pmft, radii)
      
  >       assert np.isclose(np.nanmean(rdf), 1, rtol=1e-2, atol=1e-2)
  E       assert False
  E        +  where False = <function isclose at 0x7f9dbe1970e0>(1.031277717966022, 1, rtol=0.01, atol=0.01)
  E        +    where <function isclose at 0x7f9dbe1970e0> = np.isclose
  E        +    and   1.031277717966022 = <function nanmean at 0x7f9db7e34050>(array([1.9245007 , 1.60375059, 0.68732166, 0.80187535, 1.11621046,\n       0.96225041, 0.96225035, 0.8706075 , 1.019698...    1.04344022, 1.08253169, 1.07137144, 0.96225035, 0.85424262,\n       1.12262547, 1.02239096, 1.7641257 , 1.9245007 ]))
  E        +      where <function nanmean at 0x7f9db7e34050> = np.nanmean
  
  test_pmft.py:189: AssertionError
  =========================== short test summary info ============================
  FAILED test_pmft.py::TestPMFTXYZ::test_pcf - assert False
  ============= 1 failed, 340 passed, 3 skipped in 165.52s (0:02:45) =============
```

**Comment by vyasr** [comment]
@SchoeniPhlippsn @bdice I removed the assertion because that exact check (verifying that box's compute is set correctly) is performed in many tests in freud for different compute classes and is not relevant to the main purpose of the test. Since I was already simplifying tests, I made that change as a way to further limit its scope and reduce redundancy in our tests. It's not a huge issue either way, though.

**Pull request merged**